rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.vectors/180_update_dense_vector_type.yml

```diff
@@ -0,0 +1,640 @@
+              type: dense_vector
+              dims: 4
+              index_options:
+                type: hnsw
+                m: 4
+  - do:
+      catch: /illegal_argument_exception/
+      indices.put_mapping:
+        index: test_index
+        body:
+          properties:
+            embedding:
+              type: dense_vector
+              dims: 4
+              index_options:
+                type: hnsw
+                ef_construction: 200
+  - do:
+      catch: /illegal_argument_exception/
+      indices.put_mapping:
+        index: test_index
+        body:
+          properties:
+            embedding:
+              type: dense_vector
+              dims: 4
+              index_options:
+                type: hnsw
+                ef_construction: 200
+                m: 4
+
+---
+"Disallowed dense vector update on same type but different other index_options, int8_hnsw":
+  - skip:
+      version: " - 8.13.99"
+      reason: "introduced after 8.13"
+
+  - do:
+      indices.create:
+        index: test_index
+
+  - do:
+      indices.put_mapping:
+        index: test_index
+        body:
+          properties:
+            embedding:
+              type: dense_vector
+              dims: 4
+              index_options:
+                type: int8_hnsw
+
+  - do:
+      indices.get_mapping:
+        index: test_index
+
+  - match: { test_index.mappings.properties.embedding.type: dense_vector }
+  - match: { test_index.mappings.properties.embedding.index_options.type: int8_hnsw }
+
+  - do:
+      catch: /illegal_argument_exception/
+      indices.put_mapping:
+        index: test_index
+        body:
+          properties:
+            embedding:
+              type: dense_vector
+              dims: 4
+              index_options:
+                type: int8_hnsw
+                m: 4
+  - do:
+      catch: /illegal_argument_exception/
+      indices.put_mapping:
+        index: test_index
+        body:
+          properties:
+            embedding:
+              type: dense_vector
+              dims: 4
+              index_options:
+                type: int8_hnsw
+                ef_construction: 200
+  - do:
+      catch: /illegal_argument_exception/
+      indices.put_mapping:
+        index: test_index
+        body:
+          properties:
+            embedding:
+              type: dense_vector
+              dims: 4
+              index_options:
+                type: int8_hnsw
+                confidence_interval: 0.3
+  - do:
+      catch: /illegal_argument_exception/
+      indices.put_mapping:
+        index: test_index
+        body:
+          properties:
+            embedding:
+              type: dense_vector
+              dims: 4
+              index_options:
+                type: int8_hnsw
+                ef_construction: 200
+                m: 4
+                confidence_interval: 0.3
+
+---
+"Disallowed dense vector update on same type but different other index_options, int8_flat":
```

Comment:
This is interesting indeed. I wonder if we should allow this? It seems like we should if we allow the whole type to change.

Reply:
I was a bit concerned about such fine granular tweaks but it might be reasonable to allow this too as an _advanced_ thing.

Reply:
@tteofili if this is the case, then we need to verify all parameters are the same that are shared between two types. Right now, I think you only validate the type name. For example, going from `hnsw` to `int8_hnsw`, right now, I think `m` could be adjusted and not caught.